### PR TITLE
feat(dependency): replace go-spew package 

### PR DESCRIPTION
**Pull request by dongjiang1989**
use [godebug](https://pkg.go.dev/github.com/kylelemons/godebug@v1.1.0/diff) package replace [go-spew](https://github.com/davecgh/go-spew) package 

fix: #1490

### CODE REVIEW

**Comment by dongjiang1989** [comment]
@kakkoyun  PTAL

/cc @anurag-deshpande

**Review by kakkoyun — COMMENTED** [review]
Thanks a lot for handling this.

I wonder how hard this is even without using a transitive dependency.

Could you maybe give it another shot with the standard library testing package?

**Comment by dongjiang1989** [comment]
> Thanks a lot for handling this.
> 
> I wonder how hard this is even without using a transitive dependency.
> 
> Could you maybe give it another shot with the standard library testing package?

Thanks for your review.  Hmmm... Maybe not Diff string with the standard library testing package.
 
Like:
```
-(bytes.Buffer) # HELP some_other_metric A value that represents a counter.
-# TYPE some_other_metric counter
-some_other_metric{label1="value1"} 1
+(bytes.Buffer) # HELP some_total A value that represents a counter.
+# TYPE some_total counter
+some_total{label1="value1"} 1
```

File: prometheus/testutil/testutil_test.go [hunk 1]
```diff
@@ -300,34 +300,20 @@ func TestMetricNotFound(t *testing.T) {
...
-`
-
 	err := CollectAndCompare(c, strings.NewReader(metadata+expected))
 	if err == nil {
 		t.Error("Expected error, got no error.")
 	}
 
-	if err.Error() != expectedError {
-		t.Errorf("Expected\n%#+v\nGot:\n%#+v", expectedError, err.Error())
+	if err.Error() == "" {
```
Comment (kakkoyun) [inline prometheus/testutil/testutil_test.go:315]: This is basically no-op. You are trying to pretty print an empty string.

File: prometheus/testutil/testutil_test.go [hunk 2]
```diff
@@ -300,34 +300,20 @@ func TestMetricNotFound(t *testing.T) {
...
 		},
 	})
+
 	c.Inc()
 
 	expected := `
 		some_other_metric{label1="value1"} 1
 	`
 
-	expectedError := `
```
Comment (kakkoyun) [inline prometheus/testutil/testutil_test.go:309]: We need to keep the form of this expected error. There is a reason that the original author put this here. They wanted to check for specific diff here.

**Review by kakkoyun — CHANGES_REQUESTED** [review]
It's great that the test are green now. But we are losing some information.

I'd suggest adding the following file with proper licence headers to the `testutils` package https://github.com/kylelemons/godebug/blob/master/diff/diff.go

https://pkg.go.dev/github.com/kylelemons/godebug@v1.1.0/diff

And updating the tests accordingly.

Or go back the previous version where you used `go-cmp`.

In any case, good job. You're close :)

File: prometheus/testutil/testutil_test.go (hunk 1 above)
Comment (dongjiang1989) [reply prometheus/testutil/testutil_test.go]: Got it. Fixed

**Comment by dongjiang1989** [comment]
> It's great that the test are green now. But we are losing some information.
> 
> I'd suggest adding the following file with proper licence headers to the `testutils` package https://github.com/kylelemons/godebug/blob/master/diff/diff.go
> 
> https://pkg.go.dev/github.com/kylelemons/godebug@v1.1.0/diff
> 
> And updating the tests accordingly.
> 
> Or go back the previous version where you used `go-cmp`.
> 
> In any case, good job. You're close :)

Thanks for your review. Please recheck  :)

**Review by kakkoyun — COMMENTED** [review]
Great job 👏 

Only one last thing.

> I'd suggest adding the following file with proper licence headers to the testutils package https://github.com/kylelemons/godebug/blob/master/diff/diff.go

In my previous comment, I suggested copying over the `diff.go` file from this repo rather than adding as a dependency. Remember our goal is to reduce the dependencies. Since it's a single file and simple enough. We can just copy the files (with licence headers) into the `testutils` package.

**Comment by dongjiang1989** [comment]
> Great job 👏
> 
> Only one last thing.
> 
> > I'd suggest adding the following file with proper licence headers to the testutils package https://github.com/kylelemons/godebug/blob/master/diff/diff.go
> 
> In my previous comment, I suggested copying over the `diff.go` file from this repo rather than adding as a dependency. Remember our goal is to reduce the dependencies. Since it's a single file and simple enough. We can just copy the files (with licence headers) into the `testutils` package.

Got it. Thanks. Fixed

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**